packages/chevrotain/api.d.ts

```diff
@@ -1624,6 +1624,16 @@ export interface CstNode {
      * the fullName will **also** include the name of the top level rule containing this nested rule.
      */
     readonly fullName?: string
+    location?: CstNodeLocation
+}
+
+export interface CstNodeLocation {
+    startOffset?: number
```

Comment:
does not this always have at least the offset parts?

Reply:
It is always present for the tokens, but we can disable its computation for the nodes to optimize the performance, I think

Reply:
What are our location collection modes?

1. Start + end offsets
2. Start + end line + column (I don't think it would make sense to collect lines without columns or vice versa)
3. Both 1 + 2.

We also simplify to two options.
1 offsets (End/start)
2. full.

Anyhow in that case we can define two/three interfaces for the CSTNodeLocation
each with **mandatory** (not optional properties) which will also help specify the types in the implementation.

```typescript
// e.g:
interface CSTNode {
  location: offsetsLoc || FullLoc
}
```

It is still not the easiest to use because how would an end user know which of he two/three union types
is relevant (they would have to do casting). but right now they also do not know which properties are really available... 

Do you have a better idea how to define this?